core/src/main/scala/org/apache/spark/h2o/utils/H2OSchemaUtils.scala

```diff
@@ -231,9 +231,14 @@ object H2OSchemaUtils {
     val maxCollectionSizes = if (sizeFromMetadata.forall(_.isDefined)) {
       sizeFromMetadata.map(_.get).toArray
     } else {
-      flatDataFrame.rdd.map { row =>
+      val sizes = flatDataFrame.rdd.map { row =>
         collectionIndices.map { idx => getCollectionSize(row, idx) }
-      }.reduce((a, b) => a.indices.map(i => if (a(i) > b(i)) a(i) else b(i))).toArray
+      }
+      if (sizes.isEmpty) {
```

Comment:
This is the bug for normal SparkDataFrame conversions. Will fail on empty collections